internal/driver/glfw/window_test.go

```diff
@@ -59,32 +59,26 @@ func TestGLDriver_CreateSplashWindow(t *testing.T) {
 	assert.True(t, w.centered)
 }
 
-func cursorForObject(obj fyne.CanvasObject) *glfw.Cursor {
-	if wid, ok := obj.(*widget.Entry); ok {
-		if !wid.Disabled() {
-			return entryCursor
-		}
-	} else if _, ok := obj.(*widget.Hyperlink); ok {
-		return hyperlinkCursor
-	} else if _, ok := obj.(*widget.SelectEntry); ok {
-		return entryCursor
+func cursorForObject(obj fyne.CanvasObject) fyne.Cursor {
+	cursor := fyne.DefaultCursor
+	if cursorable, ok := obj.(fyne.Cursorable); ok {
+		cursor = cursorable.Cursor()
 	}
-
-	return defaultCursor
+	return cursor
 }
 
 func TestWindow_Cursor(t *testing.T) {
 	d.CreateWindow("Test")
 
-	assert.Same(t, defaultCursor, cursorForObject(widget.NewLabel("")))
+	assert.Equal(t, fyne.DefaultCursor, cursorForObject(widget.NewLabel("")))
 
 	entry := widget.NewEntry()
-	assert.Same(t, entryCursor, cursorForObject(entry))
+	assert.Equal(t, fyne.TextCursor, cursorForObject(entry))
 	entry.Disable()
-	assert.Same(t, defaultCursor, cursorForObject(entry))
+	assert.Equal(t, fyne.TextCursor, cursorForObject(entry))
 
 	selectEntry := widget.NewSelectEntry([]string{"A", "B", "C"})
-	assert.Same(t, entryCursor, cursorForObject(selectEntry))
+	assert.Equal(t, fyne.TextCursor, cursorForObject(selectEntry))
 }
```

Comment:
This test does not really test anything on glfw.window. Instead it tests the `Cursor()` implementation of a couple of widgets.
The `Cursor()` implementation should be tested in the object specific tests (like entry_test and so on).
The window test should test the following cases:
* applied cursor for a non-cursorable object
* applied cursor for a `fyne.Cursorable` object which is _not_ `fyne.Disableable`(I suggest a minimal test object instead of an existing wigdet to not introduce a dependency from `widget` for this)
* applied cursor for a cursorable object which _is_ `fyne.Disableable` and enabled (again use a minimal test object)
* applied cursor for a disabled cursorable object

Reply:
Am not sure i understood your suggestion but i hope the new test works fine?